CalibCalorimetry/HcalPlugins/src/HcalHardcodeCalibrations.cc

```diff
@@ -37,19 +37,18 @@ namespace {
 
   const std::vector<HcalGenericDetId>& allCells (const HcalTopology& hcaltopology, bool killHE = false) {
     static std::vector<HcalGenericDetId> result;
-    int maxDepthHB=hcaltopology.maxDepthHB();
-    int maxDepthHE=hcaltopology.maxDepthHE();
+    int maxDepth=hcaltopology.maxDepth();
 
 #ifdef DebugLog
-    std::cout << std::endl << "HcalHardcodeCalibrations:   maxDepthHB, maxDepthHE = " 
-	      <<  maxDepthHB << ", " <<  maxDepthHE << std::endl;
+    std::cout << std::endl << "HcalHardcodeCalibrations:   maxDepth = " 
+	      <<  maxDepth << std::endl;
 #endif
 
     if (result.empty()) {
       for (int eta = -HcalDetId::kHcalEtaMask2; 
            eta <= (int)(HcalDetId::kHcalEtaMask2); eta++) {
         for (unsigned int phi = 0; phi <= HcalDetId::kHcalPhiMask2; phi++) {
-          for (int depth = 1; depth < maxDepthHB + maxDepthHE; depth++) {
+          for (int depth = 1; depth <= maxDepth; depth++) {
```

Comment:
I don't understand the logic here. Won't this lead to thousands of useless calls to ``hcaltopology.valid(cell)``? Why not reorder the two loops as follows:
```
          for (int det = 1; det <= HcalForward; det++) {
            HcalSubdetector maxDepth = static_cast<HcalSubdetector>(det);
            for (int depth = 1; depth <= maxDepth; depth++) {
```
This would allow you to remove the magic number ``minMaxDepth_`` introduced in this PR, as well as the the new ``int HcalTopology::maxDepth(void)`` method.